File: haskell/fizzbuzz.hs
main :: IO ()
main = printAll $ map fizzbuzz [1..30]
    where
        printAll [] = return ()
        printAll (x:xs) = putStrLn x >> printAll xs


fizzbuzz :: Integer -> String
fizzbuzz x
    | mod x 15 == 0 = "FizzBuzz"
    | mod x 5 == 0 = "Buzz"
    | mod x 3 == 0 = "Fizz"
    | otherwise = show x

File: python/doctest/fizzbuzz.py
#!/usr/bin/env python3
# coding: utf-8

__author__ = "matsub.rk <matsub.rk@gmail.com>"
__version__ = "0.1"
__date__ = "02/February/2015"

def fizzbuzz(n):
    if not isinstance(n, int):
        raise TypeError(
            '\'%s\' object cannot be interpreted as an integer'
            % type(n).__name__)
    if n < 1:
        raise ValueError("Request Natural number.")

    for i in range(1, n+1):
        echo = ''
        if i % 3 == 0:
            echo += 'fizz'
        if i % 5 == 0:
            echo += 'buzz'
        if echo == '':
            echo = str(i)
        yield echo
